types/artifacts.d.ts

```diff
@@ -589,6 +589,7 @@ declare global {
         timeOrigin: number;
         firstPaint?: number;
         firstContentfulPaint: number;
+        firstContentfulPaintAllFrames: number;
```

Comment:
```suggestion
        firstContentfulPaintAllFrames?: number;
```

if we go with the `requestOrUndefined` approach to make this throw in lantern situations :)

I also happen to think that's the right outcome to reflect the fact that this isn't really being computed 👍

Reply:
I made this optional in `TraceTimesWithoutFCP`:
https://github.com/GoogleChrome/lighthouse/blob/fbd963f7727d15556ac9720e8fcc1eac90fef344/lighthouse-core/lib/tracehouse/trace-processor.js#L19-L25

There should always be a `firstContentfulPaintAllFrames` in `traceOfTab` if there is also a `firstContentfulPaint`.

Reply:
> There should always be a firstContentfulPaintAllFrames in traceOfTab if there is also a firstContentfulPaint.

Yeah that's fine. The paradigm of the `metrics` object though is that the value without `observed` in front of it is the value of that metric with throttling taken into account. In the case of `AllFrames` under simulated throttling, we aren't actually computing it, so undefined feels appropriate even when `observedFirstContentfulPaintAllFrames` and `firstContentfulPaintAllFrames` in trace of tab can always exist.

Reply:
oh shoot now that I say that I realized I made this comment on the wrong typedef, I'm talking about `TimingSummary` down below